drivers/hmis/app/graphql/types/hmis_schema/service_input.rb

```diff
@@ -20,9 +20,10 @@ class HmisSchema::ServiceInput < Types::BaseInputObject
     argument :referral_outcome, HmisSchema::Enums::Hud::PATHReferralOutcome, required: false
 
     def to_params
-      result = to_h
+      result = to_h.except(:service_type)
 
-      result[:type_provided] = type_provided.split(':').last&.to_i if type_provided.present?
+      result[:record_type] = record_type || type_provided&.split(':')&.first&.to_i
+      result[:type_provided] = type_provided&.split(':')&.last&.to_i
       result[:sub_type_provided] = sub_type_provided.split(':').last&.to_i if sub_type_provided.present?
 
       enrollment = Hmis::Hud::Enrollment.editable_by(current_user).find_by(id: enrollment_id)
```

Comment:
Now I'm getting this error. It's fixed by removing `to_i` on line 11 of `drivers/hmis/app/graphql/mutations/create_service.rb`. I don't think we can remove that check entirely, because we are already validating the enrollment ID here.

<img width="527" alt="Screen Shot 2023-01-24 at 8 27 20 AM" src="https://user-images.githubusercontent.com/5333982/214307130-07c9eb1b-464a-4fc6-b0f0-51cfbaa136b7.png">